doc/source/whatsnew/v1.0.0.rst

```diff
@@ -226,6 +226,8 @@ with migrating existing code.
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+- Removed the previously deprecated ``raise_with_traceback`` function in ``pandas.compat`` (:issue:`29174`)
```

Comment:
it's the `assert_raises_regex` that was (semi-)public.

Reply:
Whoops...nice catch. Will update shortly